ml-algorithms/src/main/java/org/opensearch/ml/engine/algorithms/regression/LogisticRegression.java

```diff
@@ -0,0 +1,145 @@
+     */
+    public LogisticRegression(MLAlgoParams parameters) {
+        this.parameters = parameters == null ? LogisticRegressionParams.builder().build() : (LogisticRegressionParams)parameters;
+        validateParameters();
+        createObjective();
+        createOptimiser();
+    }
+
+    private void validateParameters() {
+        if (parameters.getLearningRate() != null && parameters.getLearningRate() < 0) {
+            throw new IllegalArgumentException("Learning rate should not be negative.");
+        }
+
+        if (parameters.getEpsilon() != null && parameters.getEpsilon() < 0) {
+            throw new IllegalArgumentException("Epsilon should not be negative.");
+        }
+
+        if (parameters.getEpochs() != null && parameters.getEpochs() < 0) {
+            throw new IllegalArgumentException("Epochs should not be negative.");
+        }
+
+        if (parameters.getBatchSize() != null && parameters.getBatchSize() < 0) {
+            throw new IllegalArgumentException("MiniBatchSize should not be negative.");
+        }
+    }
+
+    private void createObjective() {
+        LogisticRegressionParams.ObjectiveType objectiveType = Optional.ofNullable(parameters.getObjectiveType()).orElse(DEFAULT_OBJECTIVE_TYPE);
+        switch (objectiveType) {
+            case HINGE:
+                objective = new Hinge();
+                break;
+            default:
+                objective = new LogMulticlass();
+                break;
+        }
+    }
+
+    private void createOptimiser() {
+        LogisticRegressionParams.OptimizerType optimizerType = Optional.ofNullable(parameters.getOptimizerType()).orElse(DEFAULT_OPTIMIZER_TYPE);
+        Double learningRate = Optional.ofNullable(parameters.getLearningRate()).orElse(DEFAULT_LEARNING_RATE);
+        Double epsilon = Optional.ofNullable(parameters.getEpsilon()).orElse(DEFAULT_EPSILON);
+
+        switch (optimizerType) {
+            // ToDo: Add more possible optimizer. Tribuo only provides AdaGrad for logistic regression.
+            case ADA_GRAD:
+                optimiser = new AdaGrad(learningRate, epsilon);
+                break;
+            default:
+                //Use default SGD with a constant learning rate.
+                optimiser = new AdaGrad(learningRate, epsilon);
+                break;
+        }
+    }
+
+    @Override
+    public Model train(DataFrame dataFrame) {
+        MutableDataset<Label> trainDataset = TribuoUtil.generateDatasetWithTarget(dataFrame, new LabelFactory(),
+                "Logistic regression training data from OpenSearch", TribuoOutputType.LABEL, parameters.getTarget());
+        // Integer epochs = Optional.ofNullable(parameters.getEpochs()).orElse(DEFAULT_EPOCHS);
+        // LinearSGDTrainer(objective=LogMulticlass,optimiser=AdaGrad(initialLearningRate=1.0,epsilon=0.1,initialValue=0.0),epochs=5,minibatchSize=1,seed=12345)
+        Trainer<Label> logisticRegressionTrainer = new LogisticRegressionTrainer();
```

Comment:
Our `LogisticRegressionTrainer` is designed to be very simple for onboarding users & demos, I recommend you use `LinearSGDTrainer` and configure it given you're wrapping all the interface.

Reply:
@Craigacp Thanks for your suggestion. We have integrated `LinearSGDTrainer` to provide linear regression https://github.com/opensearch-project/ml-commons/blob/main/ml-algorithms/src/main/java/org/opensearch/ml/engine/algorithms/regression/LinearRegression.java#L203.  

You mean we should wrap `LinearSGDTrainer` rather than `LogisticRegressionTrainer` to provide logistic regression?

Reply:
Well, this is slightly unfortunate naming on our part. We have three classes all called `LinearSGDTrainer`, `org.tribuo.classification.sgd.linear.LinearSGDTrainer`, `org.tribuo.multilabel.sgd.linear.LinearSGDTrainer` and `org.tribuo.regression.sgd.linear.LinearSGDTrainer`. They have basically the same interface, but are constrained by the type system to accept the appropriate kind of objective function, and have the right generic types. `LogisticRegressionTrainer` is just a `org.tribuo.classification.sgd.linear.LinearSGDTrainer` subclass with a constructor that has fairly sensible defaults for running quick tests & demos. So for your use case I'd recommend you use `org.tribuo.classification.sgd.linear.LinearSGDTrainer` as then you can expose basically the same set of options as you do for linear regression (aside from the difference in objective functions).

Reply:
Thanks for the details. @Zhangxunmt I think we should change to `org.tribuo.classification.sgd.linear.LinearSGDTrainer`. 

BTW, @Craigacp , For linear regression `org.tribuo.regression.sgd.linear.LinearSGDTrainer` and this `org.tribuo.classification.sgd.linear.LinearSGDTrainer`, so you have any suggestion for the default parameter value like objective function, epochs, batch size etc?

Reply:
For the classification/logistic regression the ones we use in the constructor are fine - https://github.com/oracle/tribuo/blob/main/Classification/SGD/src/main/java/org/tribuo/classification/sgd/linear/LogisticRegressionTrainer.java#L44. You could use them for the linear regression too, substituting the `LogMulticlass` objective for `SquaredLoss`. For the linear models a batch size of 1 tends to work just fine, and there isn't really any performance benefit to larger ones given the way the linear algebra system is setup (as it's not multithreaded yet).

Unfortunately we don't have any convergence metrics like validation error built into Tribuo's trainers, and we'd need to modify the trainer API to enable such things.

Reply:
I was using LinearSGDTrainer with the default parameters in the first place (check the commented out line), then I realized it's almost equivalent to the given LogisticRegressionTrainer since it's basically using LinearSGDTrainer under the hood. We can change it to LinearSGDTrainer, and let customers to configure parameters through the training requests.